packages/components/src/CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# Contributing components
+
+These guidelines document how to contribute to components in the [`@wordpress/components`](https://www.npmjs.com/package/@wordpress/components) npm package, or “WordPress Components”. This can be addition, modification, or deprecation.
+
+## Does it belong in the component library?
+
+A component library should include what’s shared across many products and omit what’s not. Questions to help determine if a component should be added:
+
+- Is it relevant to other products/plugins?
+- Is there overlap with any existing components?
+- How much will it cost to make and maintain?
+- Can you articulate how the component should be used?
+
+Here’s a flowchart that can help determine if a new component is necessary:
+
+[![New component flowchart](https://wordpress.org/gutenberg/files/2019/07/New_component_flowchart.png)](https://coggle.it/diagram/WtUSrld3uAYZHsn-/t/new-ui-component/992b38cbe685d897b4aec6d0dd93cc4b47c06e0d4484eeb0d7d9a47fb2c48d94)
+
+## First steps to contributing
+
+Once you’re ready to contribute to WordPress Components, start by opening a GitHub issue. Include a detailed description in which you:
+
+- Explain the rationale
+- Detail the intended behavior
+- Clarify whether it’s a variation of an existing component, or a new asset
+- Include mockups of any fidelity (optional)
+- Include any inspirations from other products (optional)
+
+This issue will be the staging ground for the contribution and an opportunity for the community to weigh in with any suggestions. Reviewers can also check if there is overlap with an existing component, or if there is a more appropriate location for the component.
+It’s encouraged to surface works-in-progress. If you’re not able to complete all of the parts yourself, someone in the community may be able to pick up where you left off.
+
+## Parts of a component contribution
+
+1. **Provide a rationale**: Explain how your component will add value to the system and the greater product ecosystem. Be sure to include any user experience and interaction descriptions.
+2. **Create a design spec**: Create sizing and styling annotations for all aspects of the component. This spec should provide a developer with everything they need to create the design in code.
+3. **Create a Figma component**: Any new components or changes to existing components will mirror the [WordPress Components Figma library](https://www.figma.com/file/ZtN5xslEVYgzU7Dd5CxgGZwq/WordPress-Components?node-id=735%3A0), so we’ll need to update the Figma library and publish the changes. Please follow the [guidelines](https://www.figma.com/file/ZtN5xslEVYgzU7Dd5CxgGZwq/WordPress-Components?node-id=746%3A38) for contributing to the Figma libraries.
+4. **Provide usage documentation**: If the contribution adds additional behavior or expands a component’s features, we’ll need to document them with development and design guidelines. Read through existing component documentation for examples.
```

Comment:
What about having the documentation step be #2? Writing documentation, especially for new components, as a first step is a great way of thinking first of usage and application, rather than starting with visuals. Starting with documentation also helps developers and informs that process.

Reply:
We should also provide a link to a template that can be reused—probably the easiest is a Markdown file stored in the repo, but I'm open to other ideas as well.